packages/logger/__tests__/LoggerUtils.test.ts

```diff
@@ -49,7 +51,88 @@ describe("LoggerUtils tests", () => {
     });
 
     it("Should not hide notSecret operand", () => {
-        const data = LoggerUtils.censorYargsArguments({_: [], $0: "test", notSecret: "canSeeMe"});
+        const data = LoggerUtils.censorYargsArguments({ _: [], $0: "test", notSecret: "canSeeMe" });
         expect(data.notSecret).not.toContain(LoggerUtils.CENSOR_RESPONSE);
     });
+
+    describe("censoreRawData", () => {
+        const secrets = ["secret0", "secret1"];
+        let impConfigSpy = null;
+        let envSettingsReadSpy = null;
+        let secureFields = null;
+        let layersGet = null;
+        let impConfig = null;
+        beforeEach(() => {
+            jest.restoreAllMocks();
+            secureFields = jest.fn();
+            layersGet = jest.fn();
+            impConfigSpy = jest.spyOn(ImperativeConfig, "instance", "get");
+            envSettingsReadSpy = jest.spyOn(EnvironmentalVariableSettings, "read");
+            impConfig = {
+                config: {
+                    exists: true,
+                    api: {
+                        layers: { get: layersGet },
+                        secure: { secureFields }
+                    }
+                }
+            };
+        });
+
+        describe("should NOT censor", () => {
+            it("data if we are using old profiles", () => {
```

Comment:
Do we not mask because there is no "secure" array in old profiles? There was some discussion about using meta data from the profiles to determine if a property in the profile definition was defined as secure. Do we not bother with that meta data for old profiles because it would only be a half-way solution?

Reply:
> Do we not bother with that meta data for old profiles because it would only be a half-way solution?

`¯\_(ツ)_/¯`
I don't recall the exact details of the discussion, however I'm more than happy to open a new issue and finalize any details around the yaml-metadata.

Reply:
Created https://github.com/zowe/imperative/issues/706